### ISSUE TITLE
[FEAT]: Perplexica Integration

### ISSUE THREAD

**Issue by CrazyAce25** [Mintplex-Labs/anything-llm#2568]
### What would you like to see?

Just wanted to start by saying I love what you all are building and the integration with LocalAI is wonderful. One of the things I currently run in combination with LocalAI and love is Perplexica (https://github.com/ItzCrazyKns/Perplexica) and while it seems you all have built in support for Perplexity, Perplexica doesnt enjoy the same kind of turn key support. I would appreciate it if Perplexica support was added so I could easily make use of it within AnythingLLM as I'm not a fan of the centralized nature of Perplexity. Also provided below are some benefits of integrating the system, including the ability to toggle on and off use with the RAG system which I thought would be quite useful.

Benefits of Perplexica Integration
Enhanced Search Capabilities
Perplexica is known for its robust search engine capabilities, which could complement AnythingLLM's RAG (Retrieval-Augmented Generation) features. Integrating Perplexica would allow AnythingLLM to leverage Perplexica's advanced search functionality, potentially improving the accuracy and relevance of the responses generated by the LLM.
Toggle Button for RAG
Adding a toggle button to use Perplexica with and without RAG would provide users with more flexibility. This feature would enable users to choose between using the full capabilities of Perplexica's search engine along with RAG or opting for a simpler, faster response generation without RAG. This toggle would be particularly useful for scenarios where real-time responses are critical but also where detailed, context-aware responses are needed.
Document Handling and Context
Perplexica excels at handling large volumes of documents and providing context-aware responses. Integrating this functionality into AnythingLLM would allow users to upload and manage a large number of documents, including PDFs, Word documents, and other formats, which could then be used as context for the LLM. This would be especially beneficial for users who need to work with extensive knowledge bases or large collections of scripts and code files.
Other Ways Perplexica Could Benefit AnythingLLM
Vector Database Integration
Perplexica uses vector databases like SearXNG for efficient retrieval and embedding of documents. Integrating these capabilities into AnythingLLM could enhance its existing vector database support (e.g., LanceDB, Pinecone, Milvus) by providing additional options and potentially better performance in certain scenarios.
Customization and Flexibility
The integration would allow for more customization options, enabling users to choose between different search engines and retrieval methods. This flexibility would be beneficial for developers who want to tailor the system to their specific needs, whether it's for personal use or within an organization.
Community and Developer Support
Integrating Perplexica into AnythingLLM could also attract more developers and users who are familiar with Perplexica's features. This could lead to a broader community and more contributions to the project, enhancing its overall ecosystem and support.
Practical Implementation
To implement this integration, AnythingLLM could use its existing API and developer tools to connect with Perplexica's services. This would involve setting up API keys, configuring the search engine settings, and integrating the Perplexica UI elements into the AnythingLLM interface.
The issue on GitHub regarding Langchain SDK integration highlights the interest in integrating AnythingLLM with other tools and frameworks. A similar approach could be taken to integrate Perplexica, leveraging the Langchain SDK or other open standards to ensure seamless interaction between the two systems.
In summary, integrating Perplexica with AnythingLLM would significantly enhance the latter's capabilities by adding robust search engine functionality, improved document handling, and increased customization options. The ability to toggle between using RAG and not using it would provide users with greater flexibility, making the system more versatile and powerful.

**Comment by CrazyAce25** [comment]
@[timothycarambat](https://github.com/timothycarambat) When I think about Perplexica I don't really think of it in those terms simply because it utilizes LocalAI for its LLM capabilities hence I think of LocalAI as the LLM provider and as Perliexica as a front end due to my current usage.  In other words, yes  I would like to be able to utilize Perplexica as an llm provider within anything-llm which will utilize LocalAI for its LLM capabilities on the back end.

**Comment by frost19k** [comment]
@timothycarambat I think the trouble is that, when it comes to Search and Web Scraping agent capabilities, the nature of the LLM changes drastically.

For example:

<img src="https://github.com/user-attachments/assets/03c45e8e-bbd2-4b90-9b52-5574557dd1a4" width=500>

The agent doesn't know who I am but the LLM does.

I haven't used Perplexica, but I do use Perplexity - and it is able to retain my defined system prompt, context, and uploaded files in addition to web searches. 

Essentially, both @CrazyAce25 and I would prefer if AnythingLLM could do web search without losing context.
